pokerole_irl/worlds_app/views.py

```diff
@@ -47,6 +46,22 @@ def form_valid(self, form):
         membership.save()
         return response
 
+class WorldUpdateView(LoginRequiredMixin, UpdateView):
+    template_name = "world_update.html"
+    model = World
+    fields = ("__all__")
+    context_object_name = "world"
+
+    def dispatch(self, request, *args, **kwargs):
+        self.world = World.objects.get(pk=self.kwargs.get("pk"))
+        if request.user.is_superuser or self.world.user == request.user:
+            return super().dispatch(request, *args, **kwargs)
```

Comment:
This will break if accessed by an anonymous user (not logged in). You should do something like:

```suggestion
        response = super().dispatch(request, *args, **kwargs)
        if request.user.is_superuser or self.world.user == request.user:
            return response
```

Doing it like this, if accessing as an anonymous user, `super().dispatch()` will raise an exception when called.

Reply:
Actually, you should not make any checks on the world object here in the dispatch-method. Wait for the get_object-method, where you have the world-object anyway, and do the check there. It's fine to raise a PermissionDenied in that method.

Reply:
Like, the get_object method of the world and user views?

Reply:
Permission denied check is now in the get_object method